### PR TITLE
Use user defined sequence when provided

### PR DESCRIPTION
**Pull request by superkhau**
* Rename AppConfig to ApplicationOptions

* Make components and sequence keys optional in ApplicationOptions

cc @bajtos @raymondfeng @ritch

### CODE REVIEW

File: packages/core/src/application.ts [hunk 1]
```diff
@@ -113,7 +108,7 @@ export class Application extends Context {
   }
 }
 
-export interface AppConfig {
+export interface ApplicationOptions {
```
Comment (raymondfeng) [inline packages/core/src/application.ts]: Why do we need to rename it? I think `AppConfig` is good.

Comment (superkhau) [inline packages/core/src/application.ts]: Was trying to be consistent with other option naming

Comment (rashmihunt) [inline packages/core/src/application.ts]: @superkhau Can you please add atleast one test for this change?

**Comment by rashmihunt** [comment]
@superkhau I think ‘app.bind(‘sequence’).toClass(MySequence);' OR ‘app.sequence(MySequence)  (which is the way @bajtos  was suggesting in the last lb-next sync up ) are more cleaner than passing through appConfig ..I thought in that meeting we were agreeing to do app.sequence(MySequence)  which is similar to app.controller(MyController) to get auto population of the method in the IDE..

File: packages/core/src/application.ts (hunk 1 above)
Comment (superkhau) [inline packages/core/src/application.ts]: I don't think we need to test this, the IDE already catches it at dev time.

**Comment by superkhau** [comment]
>@superkhau I think ‘app.bind(‘sequence’).toClass(MySequence);' OR ‘app.sequence(MySequence) (which is the way @bajtos was suggesting in the last lb-next sync up ) are more cleaner than passing through appConfig ..I thought in that meeting we were agreeing to do app.sequence(MySequence) which is similar to app.controller(MyController) to get auto population of the method in the IDE..

Gonna do both until further notice -- this was discussed, but not updated in sandbox. I'm going to send another PR for the sugar (ie. `app.sequence(MySequence)`) in another PR. Will reference back to this PR once that one goes out.

File: packages/core/src/application.ts [hunk 2]
```diff
@@ -47,24 +47,19 @@ export class Application extends Context {
...
-    }
+    this._bindSequence(options);
 
     this.handleHttp = (req: ServerRequest, res: ServerResponse) =>
       this._handleHttpRequest(req, res);
 
     this.bind('logError').to(this._logError.bind(this));
+  }
 
+  protected _bindSequence(options?: ApplicationOptions): void {
```
Comment (raymondfeng) [inline packages/core/src/application.ts]: The options is already available as `this.options`. Not need to pass in as argument.

**Comment by superkhau** [comment]
>Shouldn't the test be showing user defined sequence like MySequence in sequence_aceeptance.ts and verify if this sequence is invoked or not?

@rashmihunt Nah, your sequence acceptance is the one that is ensuring the sequence actually gets invoked. Mine is simply verifying that it is bound/replacing the default sequence properly so we don't double test the same thing.

**Pull request merged**